solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -3,57 +3,133 @@ pragma solidity ^0.8.13;
 
 // ============ Internal Imports ============
 import {OwnableMulticall} from "../OwnableMulticall.sol";
-import {Router} from "../Router.sol";
-import {IInterchainAccountRouter} from "../../interfaces/IInterchainAccountRouter.sol";
-import {InterchainCallMessage} from "./InterchainCallMessage.sol";
+import {HyperlaneConnectionClient} from "../HyperlaneConnectionClient.sol";
+import {IRouter} from "../../interfaces/IRouter.sol";
+import {IInterchainAccountRouter} from "../../interfaces/middleware/IInterchainAccountRouter.sol";
+import {InterchainAccountMessage} from "../libs/middleware/InterchainAccountMessage.sol";
 import {MinimalProxy} from "../libs/MinimalProxy.sol";
 import {CallLib} from "../libs/Call.sol";
 import {TypeCasts} from "../libs/TypeCasts.sol";
+import {OverridableEnumerableMap} from "../libs/OverridableEnumerableMap.sol";
 
 // ============ External Imports ============
 import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
 import {Address} from "@openzeppelin/contracts/utils/Address.sol";
 import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 
 /*
- * @title Interchain Accounts Router that relays messages via proxy contracts on other chains.
- * @dev Currently does not support Sovereign Consensus (user specified Interchain Security Modules).
+ * @title A contract that allows accounts on chain A to call contracts via a
+ * proxy contract on chain B.
  */
-contract InterchainAccountRouter is Router, IInterchainAccountRouter {
+contract InterchainAccountRouter is
+    HyperlaneConnectionClient,
+    IRouter,
+    IInterchainAccountRouter
+{
+    // ============ Libraries ============
+
     using TypeCasts for address;
     using TypeCasts for bytes32;
 
+    // ============ Constants ============
+
     address internal immutable implementation;
     bytes32 internal immutable bytecodeHash;
 
+    // ============ Public Storage ============
+    OverridableEnumerableMap.UintToBytes32OverridableMap private routersMap;
+    OverridableEnumerableMap.UintToBytes32OverridableMap private ismsMap;
+
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when a default router is set for a remote domain
+     * @param domain The remote domain
+     * @param router The address of the remote router
+     */
+    event RemoteRouterEnrolled(uint32 indexed domain, bytes32 indexed router);
```

Comment:
feels like this event and 3 subsequent ones have maybe more indexed values than needed. Wondering what the motivation is behind it

e.g. here it and RemoteIsmEnrolled feels like `domain` is the only one that should be indexed

And for the overridden ones possibly just `domain` and `owner`

Reply:
I agree. Motivation was to keep the event signatures consistent with Router.sol

Reply:
The question here is do we care about consistency with Router.sol

Reply:
is there any tooling built out for Router.sol that we'd want to still be able to use here? If not, my feeling is it doesn't really matter

Reply:
Okay I'll change but keep consistent with Router.sol